redash_toolbelt/examples/refresh_dashboard.py

```diff
@@ -1,59 +1,111 @@
-try:
-	from .. import Redash, get_frontend_vals
-except ImportError:
-	from redash_toolbelt import Redash, get_frontend_vals
-
-
-# Build API client
-APIKEY = ''
-BASEURL = ''
-DASHBOARD_SLUG = ''
-redash = Redash(baseurl, apikey)
-
-# Get today's dynamic dates
-dd = get_frontend_vals()
-
-# Get a list of queries on this dashboard
-dash = redash.dashboard(slug=DASHBOARD_SLUG)
-viz_widgets = [i for i in dash['widgets'] if 'visualization' in i.keys()]
-l_query_ids = [i['visualization']['query']['id'] for i in viz_widgets]
-l_query_info = [redash._get(f'api/queries/{i}').json() for i in l_query_ids]
-
-# cycle through each query and refresh it
-#	+ queries without parameters: execute normally.
-# 	+ queries with parameters: pass the default value
-#	+ queries with dynamic date parameters: pass a calculated value
-
-# Refresh queries without parameters
-for q in [i for i in l_query_info if 'parameters' not in i['options']]:
-
-	r = redash._post(f"api/queries/{q['id']}/refresh")
-
-	print(r.status_code)
-
-# Prepare params
-for q in [i for i in l_query_info if 'parameters' in i['options']]:
-
-	raw_params = {f"p_{p.get('name')}": p.get('value') for p in q['options']['parameters'] }
-	
-	# Remove nesting of start and end values on date range parameters
-	flat_params = {}
-	for k,v in raw_params.items():
-
-		# Case: the date parameter is dynamic.
-		if v in dd._fields:
-			flat_params[f"{k}.start"] = getattr(dd, v).start.strftime('%Y-%m-%d')
-			flat_params[f"{k}.end"] = getattr(dd, v).end.strftime('%Y-%m-%d')
-
-		# Case: the date parameter is a hardcoded dictionary with 'start' and 'end' keys
-		elif isinstance(v, dict):
-			flat_params[f"{k}.start"] = v.get('start')
-			flat_params[f"{k}.end"] = v.get('end')
-		
-		# Case: this is not a date parameter
-		else:
-			flat_params.update({k:v})
-
-	r = redash._post(f"api/queries/{q['id']}/refresh", params=flat_params)
-	print(r.status_code)
-	
\ No newline at end of file
+import click
+from redash_toolbelt import Redash, get_frontend_vals
+
+
+def refresh_dashboard(baseurl, apikey, slug):
+
+    # build a client, fetch the dashboard, and calculate todays dates
+    client = Redash(baseurl, apikey)
+    todays_dates = get_frontend_vals()
+    queries_dict = get_queries_on_dashboard(client, slug)
+
+    # loop through each query and its JSON data
+    for idx, qry in queries_dict.items():
+
+        params = {}
+
+        if query_has_parameters(qry):
+            params = {
+                p.get("name"): fill_dynamic_val(todays_dates, p)
+                for p in qry["options"]["parameters"]
+            }
```

Comment:
This double assignment to `params` feels like a variant of asking, not telling.

I'd prefer to see something like 
```suggestion
        params = {
            p.get("name"): fill_dynamic_val(todays_dates, p)
            for p in qry["options"].get("parameters", [])
        }
```

(and removing `query_has_parameters`)

Reply:
Thank you! I hadn't heard of that principle before ([found explanation here](https://www.martinfowler.com/bliki/TellDontAsk.html)). Will keep in mind going forward.